packages/server/src/insights/insights.service.ts

```diff
@@ -1,42 +1,27 @@
 import { Injectable } from '@nestjs/common';
 import { Connection } from 'typeorm';
-import { InsightInterface, INSIGHTS_MAP } from './insight-classes';
-import { InsightPartial, ListInsightsResponse } from '@koh/common';
+import { Filter, InsightInterface, INSIGHTS_MAP } from './insight-objects';
+import { ListInsightsResponse } from '@koh/common';
 import { UserModel } from 'profile/user.entity';
 
-// interface generateAllInsightsParams {
-//   insights: InsightInterface<any>[];
-//   filters: any;  // TODO
-// }
+type GenerateOutputParams = {
+  insight: InsightInterface;
+  filters: Filter[];
+};
 
 @Injectable()
 export class InsightsService {
   constructor(private connection: Connection) {}
 
-  // Generate the output data for an insight by calling its compute function
-  async generateOutput({ insight, filters }): Promise<any> {
+  // Compute the output data for an insight and add it to the insight response
+  async computeOutput({
+    insight,
+    filters,
+  }: GenerateOutputParams): Promise<any> {
```

Comment:
can this be a Promise<OutputTypes>? Or whatever it was Called in common

Reply:
Yeah just made it an `Insight` type. Good catch